txscript/opcode.go

```diff
@@ -1088,6 +1100,76 @@ func opcodeCheckLockTimeVerify(op *parsedOpcode, vm *Engine) error {
 	return nil
 }
 
+// opcodeCheckSequenceVerify compares the top item on the data stack to the
+// LockTime field of the transaction containing the script signature
+// validating if the transaction outputs are spendable yet.  If flag
+// ScriptVerifyCheckSequenceVerify is not set, the code continues as if OP_NOP3
+// were executed.
+func opcodeCheckSequenceVerify(op *parsedOpcode, vm *Engine) error {
+	// If the ScriptVerifyCheckSequenceVerify script flag is not set, treat
+	// opcode as OP_NOP3 instead.
+	if !vm.hasFlag(ScriptVerifyCheckSequenceVerify) {
+		if vm.hasFlag(ScriptDiscourageUpgradableNops) {
+			return errors.New("OP_NOP3 reserved for soft-fork " +
+				"upgrades")
+		}
+		return nil
+	}
+
+	if vm.tx.Version < 2 {
+		return fmt.Errorf("invalid transaction version: %d",
+			vm.tx.Version)
+	}
+
+	// PeekByteArray is used here instead of PeekInt because we do not want
+	// to be limited to a 4-byte integer for reasons specified above.
+	so, err := vm.dstack.PeekByteArray(0)
+	if err != nil {
+		return err
+	}
+
+	stackSequence, err := makeScriptNum(so, vm.dstack.verifyMinimalData, 5)
```

Comment:
Need to be made clear why this is using 5 bytes.  I understand why, but it's not obvious from the code without a descriptive comment.